develop/components/pages/ButtonGroupPage.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { ButtonGroup, ButtonGroupSizes } from "../../../src/ButtonGroup/ButtonGroup";
+import { Button } from "../../../src/Button";
+import { CheckBox } from "../../../src/CheckBox/CheckBox";
+import Highlight from "react-highlight";
+import { RadioGroup, RadioListModel } from "../../../src/RadioGroup/RadioGroup";
+const docMD: string = require("../../../src/ButtonGroup/readme.md");
+
+const ButtonPage: React.FC = () => {
+    const [vertical, setVertical] = React.useState<boolean>(false);
+    const [hasSize, setHasSize] = React.useState<boolean>(false);
+    const [size, setSize] = React.useState<ButtonGroupSizes>(null);
+
+    const sizeCheckboxChanged: React.ChangeEventHandler<HTMLInputElement> = React.useCallback(
+        (e: React.ChangeEvent<HTMLInputElement>) => {
+            setHasSize(e.target.checked);
+            !e.target.checked && setSize(null); // Reset size
+        },
+        [setHasSize, setSize]
```

Comment:
i believe e.target.checked should be refenced in the dependecies array instead of setHasSize and SetSize

Have you ever tried this https://github.com/facebook/react/issues/14920

Reply:
You're right, this setters might be unnecessary. Let me experiment to make sure it works without any issue.

Reply:
resolved